### DISCUSSION TITLE
CDS SDK Feature Request: Allow implicit joins in LINQ queries by resolving Navigation Properties

### DISCUSSION THREAD

**Discussion by dotfede** [microsoft/PowerPlatform-DataverseServiceClient#23] [Ideas]
This has been already reported at: https://github.com/MicrosoftDocs/powerapps-docs/issues/1054

That issue pretty much sums it up.

There's also a repro project [here](https://github.com/fberasategui/cds-linq-issue-1054)

The idea is being able to simplify this:

```csharp
var accounts = 
    (from a in ctx.CreateQuery<Account>()
    join c in ctx.CreateQuery<Contact>() on a.PrimaryContactId.Id equals c.Id
    where c.FirstName.StartsWith("John")
    select a)
    .ToList();
```

into this:

```csharp
var accounts =
    ctx.CreateQuery<Account>()
       .Where(x => x.PrimaryContact.FirstName.StartsWith("John"))
       .ToList();
```

**Comment by MattB-msft** [comment]
Thanks for the feedback.. will take a look at the feasibility of this.
